README.md

```diff
@@ -34,3 +34,205 @@ Read so far up to
+        fn_computeOperands -->|accesses memory| MSM
+        fn_computeOperands -->|computes using| RC[RunContext]
+        fn_runInstruction -->|updates registers| fn_UpdateRegisterFunctions[fn_updatePc, fn_updateFp, fn_updateAp]
+        fn_UpdateRegisterFunctions -->|updates using| RC
+    end
+
+    subgraph RunContext
+        RC -->|increments PC| fn_incrementPc[fn_incrementPc]
+        fn_incrementPc -->|calls fn_add| PC[ProgramCounter]
+        RC -->|computes address| fn_computeAddress[fn_computeAddress]
+        fn_computeAddress -->|calls applyOffsetOnBaseAddress| applyOffsetOnBaseAddress
+        RC -->|computes op1 address| fn_computeOp1Address[fn_computeOp1Address]
+        fn_computeOp1Address -->|calls applyOffsetOnBaseAddress| applyOffsetOnBaseAddress
+    end
+
+    subgraph MemorySegmentManager
+        MSM -->|adds segment| fn_addSegment[fn_addSegment]
+        MSM -->|sets data in memory| fn_setData[fn_setData]
+        MSM -->|inserts value into memory| fn_insert[fn_insert]
+        fn_setData -->|calls insert_inner| fn_insert_inner
+        fn_insert -->|calls insert_inner| fn_insert_inner
+        fn_insert -->|updates segment size| UnsignedInt[unsigned integer]
+        MSM -->|manages| Memory[Memory]
+        fn_addSegment -->|calls incrementNumSegments in| Memory
+        fn_insert_inner -->|sets data in| Memory
+    end
+
+    subgraph Memory
+        Memory -->|get data on MaybeRelocatable mapping| fn_get[fn_get]
+        Memory -->|set data on MaybeRelocatable mapping| fn_set[fn_set]
+        Memory -->|increments segments| fn_incrementNumSegments[fn_incrementNumSegments]
+        Memory -->|returns number of segments| fn_getNumSegments[fn_getNumSegments]
+    end
+
+    subgraph CairoRunner
+        CR[CairoRunner] -->|initializes VM| fn_initialize[fn_initialize]
+        fn_initialize -->|initializes segments| fn_initializeSegments[fn_initializeSegments]
+        fn_initialize -->|initializes main entrypoint| fn_initializeMainEntrypoint[fn_initializeMainEntrypoint]
+        fn_initialize -->|sets initial state in VM| fn_initializeVm[fn_initializeVm]
+        fn_initializeMainEntrypoint -->|calls addSegment in VM| VM
+        fn_initializeVm -->|set initial state for Ap, Fp and Pc| VM
+        CR -->|runs VM until PC| fn_runUntilPc[fn_runUntilPc]
+        fn_runUntilPc -->|call_fn_step in VM| VM
+    end
+
+```
+- **Overview**: The Cairo VM TS Flow diagram illustrates the interconnectivity and operational dynamics of the various classes within the Cairo VM TypeScript implementation. It showcases how each class contributes to the overall functioning of the VM, detailing the interactions between the VirtualMachine, RunContext, MemorySegmentManager, Memory, and CairoRunner classes. This flowchart is a tool for understanding the architecture and inner workings of the Cairo VM, highlighting the orchestration of instruction execution, memory management, and execution control within this sophisticated virtual environment.
```

Comment:
is this chatGPT powered?

Reply:
I used Gpt to generate the initial wordings and then I rewrote/changed it myself